rust/ql/lib/codeql/rust/dataflow/internal/SsaImpl.qll

```diff
@@ -379,6 +380,12 @@ private module DataFlowIntegrationInput implements Impl::DataFlowIntegrationInpu
     none() // handled in `DataFlowImpl.qll` instead
   }
 
+  predicate allowFlowIntoUncertainDef(UncertainWriteDefinition def) {
+    exists(Variable v, BasicBlock bb, int i |
+      def.definesAt(v, bb, i) and mutablyBorrows(bb.getNode(i).getAstNode(), v)
+    )
+  }
```

Comment:
As discussed offline, perhaps this should be restricted to borrows passed into calls, i.e.

```suggestion
  predicate allowFlowIntoUncertainDef(UncertainWriteDefinition def) {
    exists(CfgNodes::CallExprBaseCfgNode call, Variable v, BasicBlock bb, int i |
      def.definesAt(v, bb, i) and
      mutablyBorrows(bb.getNode(i).getAstNode(), v)
    |
      call.getArgument(_) = bb.getNode(i)
      or
      call.(CfgNodes::MethodCallExprCfgNode).getReceiver() = bb.getNode(i)
    )
  }
```